primeqa/qg/trainers/qg_trainer.py

```diff
@@ -1,18 +1,5 @@
 from transformers import Seq2SeqTrainer
 
-
 class QGTrainer(Seq2SeqTrainer):
-
-    def __init__(self,*args,train_dataset,valid_dataset,data_collator,**kwargs):
-        """Question Generation Trainer for training and Evaluation. This class extends the Trainer class from huggingface.
-
-        Args:
-            train_dataset (Dataset): Train Dataset Generator
-            valid_dataset (Dataset): Validation Dataset Generator
-            data_collator (Dataset): Test Dataset Generator
-        """  
-        super().__init__(*args,
-                train_dataset=train_dataset,
-                eval_dataset=valid_dataset,
-                data_collator=data_collator,
-                **kwargs)
+    """ The trainer class for QG. All related functionality should go to this class. """
```

Comment:
Are we using a trainer or not?

Reply:
@jdpsen was asking to re-add it so it is there again in case someone will add some functionality for QG related training or inference.

Reply:
Don't we want to initialize our trainer here in case we have the option of introducing new params in future.
@vishwajeetkumar93 - any thoughts ?
@mxschmdt : currently it works fine, but I would still prefer a QGTrainer with its own init() here to be future ready. Will that conflict with you current changes ?

Reply:
> @mxschmdt : currently it works fine, but I would still prefer a QGTrainer with its own init() here to be future ready. Will that conflict with you current changes ?

No, that would work since I add a separate trainer, maybe we can also merge them (but the current QG implementation does the prediction/generation in the class QGmodel).
If you want to add new functionality you can always add the `init` method, or would you like to add a no-op `init` right now?